packages/utils/upgrade/resources/codemods/5.0.0/utils-public-interface.code.ts

```diff
@@ -0,0 +1,315 @@
+  'templateConfiguration',
+  'removeUndefined',
+  'getConfigUrls',
+  'getAbsoluteAdminUrl',
+  'getAbsoluteServerUrl',
+  'forEachAsync',
+];
+
+const transformImports = (root: Collection, j: JSCodeshift) => {
+  root
+    .find(j.ImportDeclaration, {
+      source: { value: '@strapi/utils' },
+    })
+    .forEach((path) => {
+      path.value.specifiers?.forEach((specifier) => {
+        if (!j.ImportSpecifier.check(specifier)) {
+          return false;
+        }
+
+        if (removed.includes(specifier.imported.name)) {
+          console.warn(
+            `Function "${specifier.imported.name}" as removed. You will have to remove it from your code.`
+          );
+
+          return false;
+        }
+      });
+
+      for (const primitive of Object.keys(changes)) {
+        const functions = Object.keys(changes[primitive]);
+
+        const specifiersToRefactor = path.value.specifiers?.filter((specifier) => {
+          return j.ImportSpecifier.check(specifier) && functions.includes(specifier.imported.name);
+        });
+
+        if (specifiersToRefactor?.length > 0) {
+          path.value.specifiers?.unshift(j.importSpecifier(j.identifier(primitive)));
+
+          specifiersToRefactor.forEach((specifier) => {
+            const index = path.value.specifiers?.indexOf(specifier);
+            path.value.specifiers?.splice(index, 1);
+          });
+        }
+      }
+
+      if (path.value.specifiers?.length === 0) {
+        j(path).remove();
+      }
+    });
+
+  root.find(j.ImportNamespaceSpecifier).forEach((specifierPath) => {
+    if (specifierPath.parent.value.source.value === '@strapi/utils') {
+      for (const primitive of Object.keys(changes)) {
+        const functions = Object.keys(changes[primitive]);
+        functions.forEach((funcName) => {
+          root
+            .find(j.CallExpression, {
+              callee: {
+                type: 'MemberExpression',
+                property: {
+                  type: 'Identifier',
+                  name: funcName,
+                },
+                object: {
+                  type: 'Identifier',
+                  name: specifierPath.value.local.name,
+                },
+              },
+            })
+            .replaceWith((path) => {
+              return j.callExpression(
+                j.memberExpression(
+                  j.memberExpression(
+                    j.identifier(specifierPath.value.local.name),
+                    j.identifier(primitive)
+                  ),
+                  j.identifier(changes[primitive][funcName])
+                ),
+                path.value.arguments
+              );
+            });
+        });
+      }
+    }
+  });
+
+  root
+    .find(j.VariableDeclarator, {
+      init: {
+        callee: {
+          name: 'require',
+        },
+        arguments: [{ value: '@strapi/utils' }],
+      },
+    })
+    .forEach((path) => {
+      // destrucured require
+      if (j.ObjectPattern.check(path.value.id)) {
```

Comment:
Shouldn't the check be done on `path.value` instead of `path.value.id`? (not sure at all, just surprised)

Reply:
yeah id is missleading it's the left operand of the variable declaration